components/event-stream/org.wso2.carbon.event.stream.ui/src/main/resources/web/eventstream/js/event_stream.js

```diff
@@ -22,6 +22,14 @@ function addEventStream(form, option, eventStreamId) {
     var eventStreamDescription = document.getElementById("eventStreamDescription").value.trim();
     var eventStreamNickName = document.getElementById("eventStreamNickName").value.trim();
 
+    if(eventStreamName.indexOf(" ") != -1){
+        CARBON.showErrorDialog("Stream name can not have spaces.");
```

Comment:
@minudika Better validate all special characters in a single place using a standard regex expression. Something similar to what is described in http://stackoverflow.com/questions/10456315/validating-alpha-numeric-values-with-all-special-characters can be used.